drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -0,0 +1,940 @@
+	LOG_DBG("source_chaining_en is %d", stream->dma_cfg.source_chaining_en);
+	LOG_DBG("dest_chaining_en is %d", stream->dma_cfg.dest_chaining_en);
+	LOG_DBG("linked_channel is %d", stream->dma_cfg.linked_channel);
+	LOG_DBG("source_data_size is %d", stream->dma_cfg.source_data_size);
+	LOG_DBG("dest_data_size is %d", stream->dma_cfg.dest_data_size);
+	LOG_DBG("source_burst_length is %d", stream->dma_cfg.source_burst_length);
+	LOG_DBG("dest_burst_length is %d", stream->dma_cfg.dest_burst_length);
+	LOG_DBG("block_count is %d", stream->dma_cfg.block_count);
+}
+
+/* This function is executed in the interrupt context */
+static void i2s_mcux_dma_tx_callback(const struct device *dma_dev, void *arg,
+				uint32_t channel, int status)
+{
+	const struct device *dev = (struct device *)arg;
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *stream = &dev_data->tx;
+	void *buffer;
+	int ret;
+
+	LOG_DBG("tx cb: %d", stream->state);
+	ret = k_msgq_get(&stream->out_queue, &buffer, K_NO_WAIT);
+	if (ret == 0) {
+		/* transmission complete. free the buffer */
+		k_mem_slab_free(stream->cfg.mem_slab, &buffer);
+	} else {
+		LOG_ERR("no buffer in output queue for channel %u", channel);
+	}
+
+	/* Received a STOP trigger, terminate TX immediately */
+	if (stream->last_block) {
+		stream->state = I2S_STATE_READY;
+		i2s_mcux_tx_stream_disable(dev, false);
+		LOG_DBG("TX STOPPED");
+		return;
+	}
+
+	switch (stream->state) {
+	case I2S_STATE_RUNNING:
+	case I2S_STATE_STOPPING:
+		/* get the next buffer from queue */
+		ret = k_msgq_get(&stream->in_queue, &buffer, K_NO_WAIT);
+		if (ret == 0) {
+			/* config the DMA */
+			i2s_mcux_config_dma_blocks(dev, I2S_DIR_TX, (uint32_t *)buffer);
+			k_msgq_put(&stream->out_queue, &buffer, K_NO_WAIT);
+			dma_start(stream->dev_dma, stream->channel);
+		}
+
+		if (ret || status) {
+			/*
+			 * DMA encountered an error (status != 0)
+			 * or
+			 * No bufers in input queue
+			 */
+			LOG_DBG("DMA status %08x channel %u k_msgq_get ret %d",
+				status, channel, ret);
+			if (stream->state == I2S_STATE_STOPPING) {
+				stream->state = I2S_STATE_READY;
+			} else {
+				stream->state = I2S_STATE_ERROR;
+			}
+			i2s_mcux_tx_stream_disable(dev, false);
+		}
+		break;
+	case I2S_STATE_ERROR:
+		i2s_mcux_tx_stream_disable(dev, true);
+		break;
+	}
+}
+
+static void i2s_mcux_dma_rx_callback(const struct device *dma_dev, void *arg,
+				uint32_t channel, int status)
+{
+	struct device *dev = (struct device *)arg;
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *stream = &dev_data->rx;
+	void *buffer;
+	int ret;
+
+	LOG_DBG("rx cb: %d", stream->state);
+
+	if (status != 0) {
+		stream->state = I2S_STATE_ERROR;
+		i2s_mcux_rx_stream_disable(dev, false);
+		return;
+	}
+
+	switch (stream->state) {
+	case I2S_STATE_STOPPING:
+	case I2S_STATE_RUNNING:
+		/* retrieve buffer from input queue */
+		ret = k_msgq_get(&stream->in_queue, &buffer, K_NO_WAIT);
+		assert(ret == 0);
```

Comment:
In Zephyr we should use `__ASSERT_NO_MSG` macro, it's located in `include/sys/__assert.h`.

Reply:
Thank you, I have changed it.